### PR TITLE
Cleans up docs/man/Dockerfile.5.md

### LINKED ISSUE #10667: Markdown syntax problems in manpages
There are errors in the manpages because parameters in angle brackets aren't included in `md2man` output. For [example][], 
```
**ADD ["<src>"... "<dest>"]** This form is required for paths containing whitespace.
```
renders as
```
ADD [""... ""] This form is required for paths containing whitespace.
```
Note the missing `<src>` and `<dest>`.
This can also be seen in GitHub's rendering of the text:

**ADD ["<src>"... "<dest>"]** This form is required for paths containing whitespace.

[example]: https://github.com/docker/docker/blame/master/docs/man/Dockerfile.5.md#L137

--------------
**PROBLEM**: man docs are often imporoperly formatted, using quotes or bolds when code blocks/segments are required

**PROPOSED SOLUTION**: man docs should be updated to use appropriate backticks for either inline or blocked code.
The `dockerfile.5.md` has already been gone through, but wouldn't hurt another set of eyes.

### PR DESCRIPTION
**Pull request by cpuguy83**
Fixes some improper usage of `**` instead of backticks
Re-arranged some things for readability.

### CODE REVIEW

**Comment by SvenDowideit** [comment]
LGTM - @fredlf @jamtur01

File: docs/man/Dockerfile.5.md [hunk 1]
```diff
@@ -29,198 +31,242 @@ A Dockerfile is similar to a Makefile.
...
-    image. The Docker daemon automatically cleans up the context it is given.
-
-Docker re-uses intermediate images whenever possible. This significantly 
-accelerates the *docker build* process.
- 
+  sudo docker build .
+
+  -- Runs the steps and commits them, building a final image
+  the path to the source repository defines where to find the context of the
+  build. The build is run by the docker daemon, not the CLI. The whole
```
Comment (fredlf) [inline docs/man/Dockerfile.5.md]: Docker

File: docs/man/Dockerfile.5.md [hunk 2]
```diff
@@ -29,198 +31,242 @@ A Dockerfile is similar to a Makefile.
...
-    succeeds. The Docker daemon runs the steps one-by-one, committing the result 
-    to a new image if necessary before finally outputting the ID of the new 
-    image. The Docker daemon automatically cleans up the context it is given.
-
-Docker re-uses intermediate images whenever possible. This significantly 
-accelerates the *docker build* process.
- 
+  sudo docker build .
+
+  -- Runs the steps and commits them, building a final image
```
Comment (fredlf) [inline docs/man/Dockerfile.5.md]: Missing period after image, capitalize "The".

**Comment by jfrazelle** [comment]
ping @lk4d4 is this not docs-review? you removed the label, I think it is, but @tiborvass would know best

File: docs/man/Dockerfile.5.md [hunk 3]
```diff
@@ -29,198 +31,242 @@ A Dockerfile is similar to a Makefile.
...
+  build. The build is run by the docker daemon, not the CLI. The whole
+  context must be transferred to the daemon. The Docker CLI reports
+  `"Sending build context to Docker daemon"` when the context is sent to the
+  daemon.
+
+  sudo docker build -t repository/tag .
+
+  -- specifies a repository and tag at which to save the new image if the build
+  succeeds. The Docker daemon runs the steps one-by-one, committing the result
+  to a new image if necessary before finally outputting the ID of the new
```
Comment (fredlf) [inline docs/man/Dockerfile.5.md]: comma after "necessary"

File: docs/man/Dockerfile.5.md [hunk 4]
```diff
@@ -29,198 +31,242 @@ A Dockerfile is similar to a Makefile.
...
+
+  -- The **FROM** instruction sets the base image for subsequent instructions. A
+  valid Dockerfile must have **FROM** as its first instruction. The image can be any
+  valid image. It is easy to start by pulling an image from the public
+  repositories.
+
+  -- **FROM** must be he first non-comment instruction in Dockerfile.
+
+  -- **FROM** may appear multiple times within a single Dockerfile in order to create
+  multiple images. Make a note of the last image id output by the commit before
```
Comment (jamtur01) [inline docs/man/Dockerfile.5.md]: ID

File: docs/man/Dockerfile.5.md [hunk 5]
```diff
@@ -29,198 +31,242 @@ A Dockerfile is similar to a Makefile.
...
+  FROM image
+
+  FROM image:tag
+
+  -- The **FROM** instruction sets the base image for subsequent instructions. A
+  valid Dockerfile must have **FROM** as its first instruction. The image can be any
+  valid image. It is easy to start by pulling an image from the public
+  repositories.
+
+  -- **FROM** must be he first non-comment instruction in Dockerfile.
```
Comment (fredlf) [inline docs/man/Dockerfile.5.md]: typo: the

File: docs/man/Dockerfile.5.md [hunk 6]
```diff
@@ -29,198 +31,242 @@ A Dockerfile is similar to a Makefile.
...
+
+    # Executable form
+    RUN ["executable", "param1", "param2"]
+
+
+  -- The RUN instruction executes any commands in a new layer on top of the current
+  image and commits the results. The committed image is used for the next step in
+  Dockerfile.
+
+  -- Layering RUN instructions and generating commits conforms to the core
```
Comment (jamtur01) [inline docs/man/Dockerfile.5.md]: **RUN**

File: docs/man/Dockerfile.5.md [hunk 7]
```diff
@@ -29,198 +31,242 @@ A Dockerfile is similar to a Makefile.
...
+  -- **RUN** has two forms:
+
+    # the command is run in a shell - /bin/sh -c
+    RUN <command>
+
+    # Executable form
+    RUN ["executable", "param1", "param2"]
+
+
+  -- The RUN instruction executes any commands in a new layer on top of the current
```
Comment (jamtur01) [inline docs/man/Dockerfile.5.md]: \*\*RUN\*\*

**Comment by jamtur01** [comment]
There a lot of variance in formatting here - some Dockerfile instructions are wrapped in *, others are not. Use of <command> versus \*\*command\*\*. I think some consistency is needed.

File: docs/man/Dockerfile.5.md [hunk 8]
```diff
@@ -29,198 +31,242 @@ A Dockerfile is similar to a Makefile.
...
- application developers with a boilerplate Dockerfile to copy-paste
- into their application is inefficient, error-prone, and
- difficult to update because it mixes with application-specific code.
- The solution is to use **ONBUILD** to register instructions in advance, to
- run later, during the next build stage.  
+  -- `ONBUILD [INSTRUCTION]`
+  The **ONBUILD** instruction adds a trigger instruction to the image, which is 
+  executed at a later time, when the image is used as the base for another
+  build. The trigger is executed in the context of the downstream build, as
+  if it had been inserted immediately after the FROM instruction in the
```
Comment (fredlf) [inline docs/man/Dockerfile.5.md]: This is not consistent. All commands/on-screen text needs to receive the same formatting.

**Comment by fredlf** [comment]
Formatting for commands is still not consistent throughout, but his is an improvement, so LGTM.

**Comment by cpuguy83** [comment]
PTAL

File: docs/man/Dockerfile.5.md [hunk 9]
```diff
@@ -29,198 +31,265 @@ A Dockerfile is similar to a Makefile.
...
+  ```
+  FROM ubuntu
+  CMD ["/usr/bin/wc","--help"]
+  ```
+
+  -- To make the container run the same executable every time, use **ENTRYPOINT** in
+  combination with **CMD**. 
+  If the user specifies arguments to  docker run, the specified commands
+  override the default in **CMD**.
+  Do not confuse **RUN** with **CMD**. RUN runs a command and commits the result.
```
Comment (fredlf) [inline docs/man/Dockerfile.5.md]: **RUN**

File: docs/man/Dockerfile.5.md [hunk 10]
```diff
@@ -29,198 +31,265 @@ A Dockerfile is similar to a Makefile.
...
-  image.
+
+  ```
+  FROM ubuntu
+  CMD ["/usr/bin/wc","--help"]
+  ```
+
+  -- To make the container run the same executable every time, use **ENTRYPOINT** in
+  combination with **CMD**. 
+  If the user specifies arguments to  docker run, the specified commands
```
Comment (fredlf) [inline docs/man/Dockerfile.5.md]: `docker run`

File: docs/man/Dockerfile.5.md [hunk 11]
```diff
@@ -29,198 +31,265 @@ A Dockerfile is similar to a Makefile.
...
- unintended consequences, because it will persist when the container is run
- interactively, as with the following command: **docker run -t -i image bash**
+  -- `ENV <key> <value>`
+  The **ENV** instruction sets the environment variable <key> to
+  the value `<value>`. This value is passed to all future 
+  RUN, **ENTRYPOINT**, and **CMD** instructions. This is
+  functionally equivalent to prefixing the command with `<key>=<value>`.  The
+  environment variables that are set with **ENV** persist when a container is run
+  from the resulting image. Use `docker inspect` to inspect these values, and
+  change them using docker run `--env <key>=<value>.`
```
Comment (fredlf) [inline docs/man/Dockerfile.5.md]: `docker run`

**Comment by fredlf** [comment]
Found a couple more, but LGTM still.

**Comment by cpuguy83** [comment]
Updated with fred's notes.

**Comment by fredlf** [comment]
Still a handful of inconsistently formatted **Dockerfile**, but we have a separate ticket to deal with this issue site-wide, so I'm going to merge this. Thanks @cpuguy83 !

**Pull request merged**